File: pricing/Token.ts
import { Address } from "../core/BaseTypes/Address.js";

class Token {
  name: string;
  decimals: bigint;
  address: Address;
  constructor(name: string, decimals: bigint, address: Address) {
    this.name = name.toUpperCase();
    this.decimals = decimals;
    this.address = address;
  }

  equals(token0: Token): boolean {
    if (this.address.equals(token0.address)) return true;
    return false;
  }
}
export default Token;

File: pricing/token.py
from core.base_types import Address


class Token:
    name: str
    decimals: int
    address: Address

    def __init__(self, name: str, decimals: int, address: Address):
        self.name = name.upper()
        self.decimals = decimals
        self.address = address

    def __eq__(self, other: Token) -> bool:
        if self.address.__eq__(other.address):
            return True
        return False
